server/queues/tasks/DetachDocumentsFromCollectionTask.test.ts

```diff
@@ -51,4 +51,30 @@ describe("DetachDraftsFromCollectionTask", () => {
     expect(draft?.deletedAt).toBe(null);
     expect(draft?.collectionId).toBe(null);
   });
+
+  it("should detach archived documents from deleted collection", async () => {
+    const collection = await buildCollection();
+    const document = await buildDocument({
+      title: "test",
+      collectionId: collection.id,
+      archivedAt: new Date(),
+      createdById: collection.createdById,
+      teamId: collection.teamId,
+    });
+    await collection.destroy({ hooks: false });
+
+    const task = new DetachDocumentsFromCollectionTask();
+    await task.perform({
+      collectionId: collection.id,
+      ip,
+      actorId: collection.createdById,
+    });
+
+    const detachedDocument = await Document.findByPk(document.id);
+    expect(detachedDocument).not.toBe(null);
+    expect(detachedDocument?.archivedAt).toBeTruthy();
+    expect(detachedDocument?.publishedAt).toBeTruthy();
+    expect(detachedDocument?.deletedAt).toBe(null);
+    expect(detachedDocument?.collectionId).toBe(null);
```

Comment:
This is interesting. We have a kinda business rule where if a document is published, it's implied that it has a parent collection(since we always publish a doc _to_ a collection). We'd have to be mindful of this case since it violates the rule. Hopefully, doesn't regress since we already filter out archived/deleted docs in most of the scenarios.